netket/optimizer/sr/sr_onthefly_logic.py

```diff
@@ -161,7 +165,18 @@ def Odagger_DeltaO_v(samples, params, v, forward_fn, vjp_fun=None):
     # v_tilde /= n_samples (elementwise):
     v_tilde = v_tilde * (1.0 / (samples.shape[0] * n_nodes))
 
-    return OH_w(samples, params, v_tilde, forward_fn, dtype=dtype, vjp_fun=vjp_fun)
+    # ⟨Ô^†ΔÔ⟩v
+    res = OH_w(samples, params, v_tilde, forward_fn, dtype=dtype, vjp_fun=vjp_fun)
+
+    if centered:
+        # ⟨ΔÔ⟩v; is a scalar close to 0; (MPI sum):
+        v_tilde_sum = _sum(v_tilde)
```

Comment:
sorry it's not clear to me why you need to center it here explicitly, don't you already have a centered quantity when you compute the first product?

Reply:
Yes but only for O_l, not for O_k
I am not aware of a simple way to do `(jacobian + matrix) @ vector` as a jvp directly with jax (where the matrix is ⟨Ô⟩ broadcast to all columns)